qiita_pet/handlers/study_handlers/vamps_handlers.py

```diff
@@ -36,15 +33,26 @@ def display_template(self, preprocessed_data_id, msg, msg_level):
             if user.level != 'admin':
                 raise HTTPError(403, "No permissions of admin, "
                                      "get/VAMPSSubmitHandler: %s!" % user.id)
-
-        prep_template = PrepTemplate(preprocessed_data.prep_template)
-        sample_template = SampleTemplate(preprocessed_data.study)
-        study = Study(preprocessed_data.study)
+        prep_templates = preprocessed_data.prep_templates
+        allow_submission = len(prep_templates) == 1
+        msg_list = ["Submission to EBI disabled:"]
+        if not allow_submission:
+            msg_list.append(
+                "Only artifacts with a single prep template can be submitted")
+        # If allow_submission is already false, we technically don't need to
+        # do the following work. However, there is no clean way to fix this
+        # using the current structure, so we perform the work as we
+        # did so it doesn't fail.
+        # We currently support only one prep template for submission, so
+        # grabbing the first one
+        prep_template = prep_templates[0]
```

Comment:
Noted, thanks!

On (Mar-02-16|12:57), Antonio Gonzalez wrote:
> > -        prep_template = PrepTemplate(preprocessed_data.prep_template)
> > -        sample_template = SampleTemplate(preprocessed_data.study)
> > -        study = Study(preprocessed_data.study)
> > +        prep_templates = preprocessed_data.prep_templates
> > +        allow_submission = len(prep_templates) == 1
> > +        msg_list = ["Submission to EBI disabled:"]
> > +        if not allow_submission:
> > +            msg_list.append(
> > +                "Only artifacts with a single prep template can be submitted")
> > +        # If allow_submission is already false, we technically don't need to
> > +        # do the following work. However, there is no clean way to fix this
> > +        # using the current structure, so we perform the work as we
> > +        # did so it doesn't fail.
> > +        # We currently support only one prep template for submission, so
> > +        # grabbing the first one
> > +        prep_template = prep_templates[0]
> 
> Theoretically it could but in practice it can't: the only way to get to this point is by clicking on a button that is only shown when you see the summary of the prep template ... 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiita/pull/1671/files#r54789962